vanetza/geonet/router.cpp

```diff
@@ -229,8 +235,14 @@ DataConfirm Router::request(const ShbDataRequest& request, DownPacketPtr payload
             pdu->extended().source_position = m_local_position_vector;
 
             // step 2: encapsulate packet by security
-            if (m_mib.itsGnSecurity) {
-                payload = encap_packet(ctrl.its_aid, *pdu, std::move(payload));
+            if (m_mib.itsGnSecurity && ctrl.security_profile) {
+                auto encap_result = encap_packet(ctrl.security_profile, ctrl.its_aid, *ctrl.permissions, *pdu, std::move(payload));
+                if (encap_result) {
+                    payload = std::move(encap_result.get());
+                } else {
+                    // TODO: How to throw an error here?
```

Comment:
As far as I know, no error reporting is foreseen at this point by specifications.
We may invoke a hook here so custom error logging can be realised without hassle.